### PR TITLE
Feat: DO Redis

### PR DESCRIPTION
**Pull request by ahmedhesham6**
- Add DO Schema Helper
- Update Redis version Type
- AddRedisCluster transformer
- AddRedisClusterFirewallRules transformer

### CODE REVIEW

File: v1/traits/kv.cue [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 	$metadata: traits: Redis: null
 	redis: {
 		name:    string | *$metadata.id
-		version: string | *"7.0"
+		version: string | int | *"7.0"
```
Comment (kajogo777) [inline v1/traits/kv.cue:12]: I think this should stay a string, to support semantic versioning

File: v1/transformers/terraform/digitalocean/helpers/schema.cue [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	size:      *"db-s-1vcpu-1gb" | "db-s-1vcpu-2gb" | "db-s-2vcpu-4gb" | "db-s-4vcpu-8gb" | "db-s-6vcpu-16gb" | "db-s-8vcpu-32gb" | "db-s-16vcpu-64gb" | "gd-2vcpu-8gb" | "gd-4vcpu-16gb" | "gd-8vcpu-32gb" | "gd-16vcpu-64gb" | "gd-32vcpu-128gb" | "gd-40vcpu-160gb" | "so1_5-2vcpu-16gb" | "so1_5-4vcpu-32gb" | "so1_5-8vcpu-64gb" | "so1_5-16vcpu-128gb" | "so1_5-24vcpu-192gb" | "so1_5-32vcpu-256gb" | "m-2vcpu-16gb" | "m-4vcpu-32gb" | "m-8vcpu-64gb" | "m-16vcpu-128gb" | "m-24vcpu-192gb" | "m-32vcpu-256gb"
+	nodeCount: int | *1
+}
+
+#DatabaseFirewallRule: {
+	kubernetes: name: string
+} | {
+	droplet: name: string
+} | {
+	ip: list.Repeat([ uint8], 4)
```
Comment (kajogo777) [inline v1/transformers/terraform/digitalocean/helpers/schema.cue:18]: `ip: net.IP` from package `net`

File: v1/traits/kv.cue (hunk 1 above)
Comment (kajogo777) [reply v1/traits/kv.cue:12]: if you need an int you can use `strconv.Atoi(string)` to convert the string to int
you can find the list of available cue helper functions here https://pkg.go.dev/cuelang.org/go/pkg

File: v1/transformers/terraform/digitalocean/redis.cue [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	"guku.io/devx/v1/traits"
+	schema "guku.io/devx/v1/transformers/terraform"
+	helpers "guku.io/devx/v1/transformers/terraform/digitalocean/helpers"
+	"strconv"
+)
+
+#AddRedisCluster: v1.#Transformer & {
+	traits.#Redis
+	redis: _
+	redis: version: <= "7" & >= "6"
```
Comment (ahmedhesham6) [inline v1/transformers/terraform/digitalocean/redis.cue:14]: Not sure that will work with "7.0"

**Review by kajogo777 — APPROVED** [review]

**Pull request merged**